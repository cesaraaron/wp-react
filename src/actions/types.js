export const GET_ENDPOINT = 'GET_ENDPOINT'

export const FETCH_POSTS_REQUEST = 'FETCH_POSTS_REQUEST'

export const FETCH_POSTS_SUCCESS = 'FETCH_POSTS_SUCCESS'

export const FETCH_POSTS_FAILURE = 'FETCH_POSTS_FAILURE'

export const FETCH_SINGLE_REQUEST = 'FETCH_SINGLE_REQUEST'

export const FETCH_SINGLE_SUCCESS = 'FETCH_SINGLE_SUCCESS'

export const FETCH_SINGLE_FAILURE = 'FETCH_SINGLE_FAILURE'

export const posts = 'posts'

export const single = 'single'
