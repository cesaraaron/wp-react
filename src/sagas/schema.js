import { schema } from 'normalizr'

export const post = new schema.Entity('post')

export const arrayOfPosts = new schema.Array(post)
