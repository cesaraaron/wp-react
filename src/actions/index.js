import WPAPI from 'wpapi'
import { normalize } from 'normalizr'
import { arrayOfPosts } from './schema'
import * as selectors from '../reducers'
import * as types from './types'

// For some reason this doesn't work when fetching http://wptest.io/demo
// I got an error ‘res is null’ or something like that.
// Had to use a local wp installation to make it work :c
const getAll = req =>
  req.then(res => {
    if (!res._paging || !res._paging.next) {
      return res
    }
    return Promise.all([res, getAll(res._paging.next)]).then(data =>
      data.reduce((prev, next) => [...prev, ...next])
    )
  })

const createOnFetch = (type, doFetching) => (dispatch, getState) => {
  const onFetchVars = types.createOnFetchVars(type)
  const state = getState()
  const endpoint = selectors.getEndpoint(state)
  const isFetching = selectors.getIsFetching(state, type)

  if (isFetching) {
    return Promise.resolve()
  }

  const api = new WPAPI({ endpoint })
  dispatch({ type: onFetchVars.request })

  return doFetching(api, dispatch)
}

export const getEndpoint = location => ({
  type: types.GET_ENDPOINT,
  location
})

export const fetchPostsByCategorySlug = (categorySlug, pageNumber) =>
  createOnFetch(types.postsByCategory, (api, dispatch) =>
    api
      .categories()
      .slug(categorySlug)
      .then(cats => {
        if (cats.length > 0) {
          return api
            .posts()
            .categories(cats[0].id)
            .page(pageNumber)
        } else {
          dispatch({
            type: types.FETCH_POSTS_BY_CATEGORY_FAILURE,
            message: '404 not found'
          })
        }
      })
      .then(
        res => {
          const response = normalize(res, arrayOfPosts)
          response._paging = res._paging

          dispatch({
            type: types.FETCH_POSTS_BY_CATEGORY_SUCCESS,
            response,
            pageNumber
          })
        },
        err =>
          dispatch({
            type: types.FETCH_POSTS_BY_CATEGORY_FAILURE,
            message: err.message
          })
      )
  )

export const fetchPostsByPageNumber = pageNumber =>
  createOnFetch(types.posts, (api, dispatch) =>
    api
      .posts()
      .page(pageNumber)
      .then(
        res => {
          const response = normalize(res, arrayOfPosts)
          response._paging = res._paging

          dispatch({
            type: types.FETCH_POSTS_SUCCESS,
            response,
            pageNumber
          })
        },
        err =>
          dispatch({ type: types.FETCH_POSTS_FAILURE, message: err.message })
      )
  )

export const fetchSingleBySlug = slug =>
  createOnFetch(types.single, (api, dispatch) =>
    api
      .posts()
      .slug(slug)
      .then(
        res =>
          dispatch({
            type: types.FETCH_SINGLE_SUCCESS,
            response: normalize(res, arrayOfPosts)
          }),
        err =>
          dispatch({ type: types.FETCH_SINGLE_FAILURE, message: err.message })
      )
  )

export const fetchCommentsByPostId = postId =>
  createOnFetch(types.comments, (api, dispatch) =>
    api
      .comments()
      .post(postId)
      .then(
        res =>
          dispatch({
            type: types.FETCH_COMMENTS_SUCCESS,
            response: normalize(res, arrayOfPosts)
          }),
        err =>
          dispatch({ type: types.FETCH_COMMENTS_FAILURE, message: err.message })
      )
  )

export const fetchAllCategories = () =>
  createOnFetch(types.allCategories, (api, dispatch) =>
    getAll(api.categories()).then(
      res =>
        dispatch({
          type: types.FETCH_ALL_CATEGORIES_SUCCESS,
          response: normalize(res, arrayOfPosts)
        }),
      err =>
        dispatch({
          type: types.FETCH_ALL_CATEGORIES_FAILURE,
          message: err.message
        })
    )
  )
