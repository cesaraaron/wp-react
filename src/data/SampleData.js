export const posts = [
  {
    id: 1,
    date: '2017-09-15T17:42:48',
    date_gmt: '2017-09-15T17:42:48',
    guid: { rendered: 'http://localhost/?p=1' },
    modified: '2017-09-15T17:42:48',
    modified_gmt: '2017-09-15T17:42:48',
    slug: 'hello-world',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/hello-world',
    title: { rendered: 'Hello world!' },
    content: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    author: 1,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [1],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/1'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 1031,
    date: '2013-03-15T17:23:27',
    date_gmt: '2013-03-15T22:23:27',
    guid: { rendered: 'http://wptest.io/demo/?p=1031' },
    modified: '2013-03-15T17:23:27',
    modified_gmt: '2013-03-15T22:23:27',
    slug: 'tiled-gallery',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/tiled-gallery',
    title: { rendered: 'Tiled Gallery' },
    content: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery.</p>\n<p>You can install <a title="Jetpack for WordPress" href="http://jetpack.me/" target="_blank">Jetpack</a> or <a title="Slim Jetpack" href="http://wordpress.org/extend/plugins/slimjetpack/" target="_blank">Slim Jetpack</a> to test it out.</p>\n\n\t\t<style type=\'text/css\'>\n\t\t\t#gallery-1 {\n\t\t\t\tmargin: auto;\n\t\t\t}\n\t\t\t#gallery-1 .gallery-item {\n\t\t\t\tfloat: left;\n\t\t\t\tmargin-top: 10px;\n\t\t\t\ttext-align: center;\n\t\t\t\twidth: 25%;\n\t\t\t}\n\t\t\t#gallery-1 img {\n\t\t\t\tborder: 2px solid #cfcfcf;\n\t\t\t}\n\t\t\t#gallery-1 .gallery-caption {\n\t\t\t\tmargin-left: 0;\n\t\t\t}\n\t\t\t/* see gallery_shortcode() in wp-includes/media.php */\n\t\t</style>\n\t\t<div id=\'gallery-1\' class=\'gallery galleryid-1031 gallery-columns-4 gallery-size-thumbnail\'><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/fight-club\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/fight-club-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Fight Club" srcset="http://localhost:8080/wp-content/uploads/2013/03/fight-club-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/fight-club-1200x1200.jpg 1200w, http://localhost:8080/wp-content/uploads/2013/03/fight-club-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/the-dark-knight-rises\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="The Dark Knight Rises" srcset="http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/the-dark-knight-rises-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/captain-america\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/captain-america-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Captain America" srcset="http://localhost:8080/wp-content/uploads/2013/03/captain-america-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/captain-america-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/man-of-steel\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Man Of Steel" srcset="http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/man-of-steel-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><br style="clear: both" /><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/spider-man\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/spider-man-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="The Amazing Spider Man" srcset="http://localhost:8080/wp-content/uploads/2013/03/spider-man-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/spider-man-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl><dl class=\'gallery-item\'>\n\t\t\t<dt class=\'gallery-icon portrait\'>\n\t\t\t\t<a href=\'http://localhost:8080/tiled-gallery/ironman-2\'><img width="150" height="150" src="http://localhost:8080/wp-content/uploads/2013/03/ironman-2-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="Iron Man 2" srcset="http://localhost:8080/wp-content/uploads/2013/03/ironman-2-150x150.jpg 150w, http://localhost:8080/wp-content/uploads/2013/03/ironman-2-100x100.jpg 100w" sizes="(max-width: 150px) 100vw, 150px" /></a>\n\t\t\t</dt></dl>\n\t\t\t<br style=\'clear: both\' />\n\t\t</div>\n\n<p>This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery. You can install Jetpack or Slim Jetpack to test it out. This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    author: 6,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [15, 16, 17],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1031' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/6'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1031'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1031/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1031' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1031'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1031'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]
