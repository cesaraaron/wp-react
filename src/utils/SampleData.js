export const posts = [
  {
    id: 1,
    date: '2017-09-15T17:42:48',
    date_gmt: '2017-09-15T17:42:48',
    guid: { rendered: 'http://localhost/?p=1' },
    modified: '2017-09-15T17:42:48',
    modified_gmt: '2017-09-15T17:42:48',
    slug: 'hello-world',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/hello-world',
    title: { rendered: 'Hello world!' },
    content: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>\n',
      protected: false
    },
    author: 1,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [1],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/1'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 1031,
    date: '2013-03-15T17:23:27',
    date_gmt: '2013-03-15T22:23:27',
    guid: { rendered: 'http://wptest.io/demo/?p=1031' },
    modified: '2013-03-15T17:23:27',
    modified_gmt: '2013-03-15T22:23:27',
    slug: 'tiled-gallery',
    status: 'publish',
    type: 'post',
    link: 'http://localhost:8080/tiled-gallery',
    title: { rendered: 'Tiled Gallery' },
    content: {
      rendered: '<p>titled gallery content</p>',
      protected: false
    },
    excerpt: {
      rendered:
        '<p>This is a test for Jetpack&#8217;s Tiled Gallery. You can install Jetpack or Slim Jetpack to test it out. This is some text after the Tiled Gallery just to make sure that everything spaces nicely.</p>\n',
      protected: false
    },
    author: 6,
    featured_media: 0,
    comment_status: 'open',
    ping_status: 'open',
    sticky: false,
    template: '',
    format: 'standard',
    meta: [],
    categories: [15, 16, 17],
    tags: [],
    _links: {
      self: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts/1031' }],
      collection: [{ href: 'http://localhost:8080/wp-json/wp/v2/posts' }],
      about: [{ href: 'http://localhost:8080/wp-json/wp/v2/types/post' }],
      author: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/users/6'
        }
      ],
      replies: [
        {
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/comments?post=1031'
        }
      ],
      'version-history': [
        { href: 'http://localhost:8080/wp-json/wp/v2/posts/1031/revisions' }
      ],
      'wp:attachment': [
        { href: 'http://localhost:8080/wp-json/wp/v2/media?parent=1031' }
      ],
      'wp:term': [
        {
          taxonomy: 'category',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/categories?post=1031'
        },
        {
          taxonomy: 'post_tag',
          embeddable: true,
          href: 'http://localhost:8080/wp-json/wp/v2/tags?post=1031'
        }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]

export const comments = [
  {
    id: 2,
    post: 1077,
    parent: 0,
    author: 0,
    author_name: 'Chris Ames',
    author_url: 'http://chrisam.es/',
    date: '2013-03-15T18:16:59',
    date_gmt: '2013-03-15T23:16:59',
    content: {
      rendered:
        '<p>Ello! Pretend you&#8217;re reading this comment with an English accent.</p>\n'
    },
    link: 'http://wptest.io/demo/page-comments/#comment-2',
    status: 'approved',
    type: 'comment',
    author_avatar_urls: {
      '24':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=24&d=retro&r=g',
      '48':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=48&d=retro&r=g',
      '96':
        'http://2.gravatar.com/avatar/25df3939b2e33bd19783411afd5bc6e3?s=96&d=retro&r=g'
    },
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments/2' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments' }],
      up: [
        {
          embeddable: true,
          post_type: 'page',
          href: 'http://wptest.io/demo/wp-json/wp/v2/pages/1077'
        }
      ],
      children: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/comments?parent=2' }
      ]
    }
  },
  {
    id: 31,
    post: 877,
    parent: 0,
    author: 0,
    author_name: 'Jared Erickson',
    author_url: 'http://jarederickson.com/',
    date: '2013-03-14T13:07:19',
    date_gmt: '2013-03-14T18:07:19',
    content: {
      rendered:
        '<p>Non-breaking text is my favorite!</p>\n<p>Super/Duper/Long/NonBreaking/Path/Name/To/A/File/That/Is/Way/Deep/Down/In/Some/Mysterious/Remote/Desolate/Part/Of/The/Operating/System/To/A/File/That/Just/So/Happens/To/Be/Strangely/Named/Supercalifragilisticexpialidocious.txt</p>\n'
    },
    link: 'http://wptest.io/demo/2013/01/05/non-breaking-text/#comment-31',
    status: 'approved',
    type: 'comment',
    author_avatar_urls: {
      '24':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=24&d=retro&r=g',
      '48':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=48&d=retro&r=g',
      '96':
        'http://2.gravatar.com/avatar/b2c1febfd11117eef66c351c1d4c10f1?s=96&d=retro&r=g'
    },
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments/31' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/comments' }],
      up: [
        {
          embeddable: true,
          post_type: 'post',
          href: 'http://wptest.io/demo/wp-json/wp/v2/posts/877'
        }
      ],
      children: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/comments?parent=31' }
      ]
    }
  }
]

export const categories = [
  {
    id: 2,
    count: 3,
    description: 'Posts in this category test image and text alignment.',
    link: 'http://wptest.io/demo/category/alignment/',
    name: 'Alignment',
    slug: 'alignment',
    taxonomy: 'category',
    parent: 0,
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories/2' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories' }],
      about: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/taxonomies/category' }
      ],
      'wp:post_type': [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/posts?categories=2' }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  },
  {
    id: 3,
    count: 2,
    description: 'Posts in this category test the aside post format.',
    link: 'http://wptest.io/demo/category/post-format-aside/',
    name: 'Aside',
    slug: 'post-format-aside',
    taxonomy: 'category',
    parent: 0,
    meta: [],
    _links: {
      self: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories/3' }],
      collection: [{ href: 'http://wptest.io/demo/wp-json/wp/v2/categories' }],
      about: [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/taxonomies/category' }
      ],
      'wp:post_type': [
        { href: 'http://wptest.io/demo/wp-json/wp/v2/posts?categories=3' }
      ],
      curies: [{ name: 'wp', href: 'https://api.w.org/{rel}', templated: true }]
    }
  }
]
