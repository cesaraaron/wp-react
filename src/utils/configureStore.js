import rootReducer from '../reducers'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { createLogger } from 'redux-logger'

export default () => {
  const logger = createLogger()

  const store = createStore(rootReducer, applyMiddleware(thunk, logger))

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers')
      store.replaceReducer(nextRootReducer)
    })
  }

  return store
}
