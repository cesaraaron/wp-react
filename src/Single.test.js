import React from 'react'
import renderer from 'react-test-renderer'
import SingleContainer from './Single'
import { posts } from './utils/SampleData'
import rootReducer from './reducers'
import { MemoryRouter, Route } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { normalize } from 'normalizr'
import { arrayOfPosts } from './actions/schema'
import * as types from './actions/types'

describe('<SingleContainer />', () => {
  const { entities, result } = normalize(posts, arrayOfPosts)
  const pageNumber = 1 // the default pageNumber is one if `match.params.pageNumber` is falsey
  const preloadedState = {
    postsById: entities.post,
    [types.SINGLE]: { idsByPage: { [pageNumber]: result } }
  }
  const store = createStore(rootReducer, preloadedState)
  store.dispatch = () => {}

  it('should render without errors', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/hello-world']} initialIndex={0}>
        <Provider store={store}>
          <Route path="/:slug" component={SingleContainer} />
        </Provider>
      </MemoryRouter>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
