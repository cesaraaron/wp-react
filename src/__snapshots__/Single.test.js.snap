// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleContainer /> should render without errors 1`] = `
<div>
  <div>
    <div>
      <article>
        <h4>
          Hello world!
        </h4>
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Welcome to WordPress. This is your first post. Edit or delete it, then start writing!</p>
",
            }
          }
        />
      </article>
    </div>
  </div>
  <br />
  <h2>
    Comments
  </h2>
</div>
`;
